### PR TITLE
Redeploy vm using python sdk

### PR DESCRIPTION
**Pull request by persiaAziz**
Please note that we will need to export the Azure credentials like we do for the deployment: Note how the code is retrieving the credentials:

subscription_id = os.environ.get('AZURE_SUBSCRIPTION_ID')
    credentials = ServicePrincipalCredentials(
            client_id=os.environ['AZURE_CLIENT_ID'],
            secret=os.environ['AZURE_SECRET'],
            tenant=os.environ['AZURE_TENANT']
        )

### CODE REVIEW

File: deploy/vm/python-scripts/redeploy-vm.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    request = urllib2.Request(url=mdUrl, headers=header)
+    response = urllib2.urlopen(request)
+    data = response.read()
+    metaData = data.decode("utf-8")
+    vm_meta_json = json.loads(metaData)
+    print(vm_meta_json)
+    GROUP_NAME = vm_meta_json["resourceGroupName"]
+    VM_NAME = sys.argv[1]
+
+    sched_event_url = "http://169.254.169.254/metadata/scheduledevents?api-version=2017-08-01"
```
Comment (pabowers) [inline deploy/vm/python-scripts/redeploy-vm.py]: What url is this?

File: deploy/vm/python-scripts/redeploy-vm.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import sys
+import urllib2
+import copy
+import socket
+
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.compute import ComputeManagementClient
+
+def main():
+    mdUrl = "http://169.254.169.254/metadata/instance/compute?api-version=2017-08-01"
```
Comment (pabowers) [inline deploy/vm/python-scripts/redeploy-vm.py]: this url is in here twice. Consider making it a constant

Comment (tniek) [reply deploy/vm/python-scripts/redeploy-vm.py]: VM metadata service (IMDS)

File: deploy/vm/python-scripts/redeploy-vm.py (hunk 1 above)
Comment (tniek) [reply deploy/vm/python-scripts/redeploy-vm.py]: VM metadata service (IMDS)

File: deploy/vm/python-scripts/redeploy-vm.py (hunk 2 above)
Comment (persiaAziz) [reply deploy/vm/python-scripts/redeploy-vm.py]: The urls are actually calling different resources: shed events and metada

File: deploy/vm/python-scripts/redeploy-vm.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+'''
+this script redeploys a VM that resides in the same azure space (i.e., resourrce group and VNet).
```
[the remark this answers is not in the document]
Comment (persiaAziz) [reply deploy/vm/python-scripts/redeploy-vm.py]: done

File: deploy/vm/python-scripts/redeploy-vm.py (hunk 2 above)
Comment (pabowers) [reply deploy/vm/python-scripts/redeploy-vm.py]: ok, the root was the same, sorry I didn't catch that difference.

**Pull request merged**